src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingTrackEmailTransactionClickCommand.cs

```diff
@@ -0,0 +1,157 @@
+            await context.SaveChangesAsync();
+
+            // Act
+            var result = await handler.Handle(command, CancellationToken.None);
+
+            // Assert
+            result.Should().Match<TrackEmailTransactionClickResponse>(p =>
+                p.FeedbackTransactionId == command.FeedbackTransactionId &&
+                p.ClickStatus == ClickStatus.Invalid);
+        }
+
+
+        [Test, AutoMoqData]
+        public async Task And_ApprenticeFeedbackTargetIdDoesMatch_Then_ReturnClickStatusValid(
+           TrackEmailTransactionClickCommand command,
+           [Frozen(Matching.ImplementedInterfaces)] ApprenticeFeedbackDataContext context,
+           TrackEmailTransactionClickCommandHandler handler
+           )
+        {
+            // Arrange
+            var feedbackTarget = new ApprenticeFeedbackTarget()
+            {
+                Id = command.ApprenticeFeedbackTargetId,
+            };
+
+            var feedbackTransaction = new FeedbackTransaction()
+            {
+                Id = command.FeedbackTransactionId,
+                ApprenticeFeedbackTargetId = feedbackTarget.Id
+            };
+
+            context.Add(feedbackTarget);
+            context.Add(feedbackTransaction);
+            await context.SaveChangesAsync();
+
+            // Act
+            var result = await handler.Handle(command, CancellationToken.None);
+
+            // Assert
+            result.Should().Match<TrackEmailTransactionClickResponse>(p =>
+                p.FeedbackTransactionId == command.FeedbackTransactionId &&
+                p.ClickStatus == ClickStatus.Valid);
+        }
+
+        [Test, AutoMoqData]
+        public async Task And_ApprenticeFeedbackTargetIdDoesMatch_Then_CreatesFeedbackTransactionClick(
+           TrackEmailTransactionClickCommand command,
+           [Frozen(Matching.ImplementedInterfaces)] ApprenticeFeedbackDataContext context,
+           TrackEmailTransactionClickCommandHandler handler
+           )
+        {
+            // Arrange
+            var feedbackTarget = new ApprenticeFeedbackTarget()
+            {
+                Id = command.ApprenticeFeedbackTargetId,
+            };
+
+            var feedbackTransaction = new FeedbackTransaction()
+            {
+                Id = command.FeedbackTransactionId,
+                ApprenticeFeedbackTargetId = feedbackTarget.Id
+            };
+
+            context.Add(feedbackTarget);
+            context.Add(feedbackTransaction);
+            await context.SaveChangesAsync();
+
+            command.LinkName = "SomeLinkName";
+            command.LinkUrl = "https://somelink.com";
+            command.ClickedOn = DateTime.Now;
+
+            // Act
+            await handler.Handle(command, CancellationToken.None);
+
+            // Assert
+            context.FeedbackTransactionClicks.Should().HaveCount(1);
+            context.FeedbackTransactionClicks.First().Should().BeEquivalentTo(new FeedbackTransactionClick
```

Comment:
If the `First()` on line 139 becomes a `Single()`, then line 138 can be deleted

Reply:
Changed to Single